Make test helpers more reliable when using S3

labkey-tchad opened the pull request:
#### Rationale
Some test helpers assume particular pages will run quickly. The aren't as reliable when particular API calls have to talk to S3 before a page is ready.

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/2541

#### Changes
* Wait for checkboxes to appear on `StartImportPage`
* Wait for file browser to be fully loaded

labkey-tchad committed:
Try to make file browser more reliable with S3

labkey-tchad committed:
Wait for checkbox

labkey-tchad committed:
Revert part of change

labkey-tchad committed:
Update StartImportPage

labkey-tchad committed:
Fix element cache

labkey-chrisj commented on src/org/labkey/test/tests/AdvancedImportOptionsTest.java:
```diff
@@ -76,7 +76,7 @@
     protected String getProjectName()
     {
-        return "AdvancedImportOptions";
+        return null;
```
interesting, this deviates from the usual pattern- maybe helpful to leave a comment/note why?

labkey-tchad committed:
Handle superclasses

labkey-tchad committed:
Update 'DeferredErrorCollectorWrapper'
Don't clear recordable error types
Don't use nested classes
Rename 'recordResults' to 'reportResults' to not confuse meaning of "record"
Add public 'reportError' to record arbitrary Throwables

labkey-tchad committed:
Merge remote-tracking branch 'origin/develop' into fb_s3StudyImport

labkey-tchad committed:
Merge remote-tracking branch 'origin/develop' into fb_s3StudyImport

labkey-tchad committed:
Allow other tests to add file browser columns

labkey-tchad committed:
Navigate to correct project

labkey-tchad committed:
Don't cleanup projects outside of `doCleanup`

labkey-tchad committed:
Remove unnecessary parameter from folder import helper
